### DISCUSSION TITLE
RequestAccess doesn't continue

### DISCUSSION THREAD

**Discussion by stefandenhartog** [shinyorg/shiny#1545] [Q&A]
I'm trying to implement push notifications in my app using Shiny.
With this instructions:
https://shinylib.net/client/push/

They work properly on Android, however on IOS I get the issue that the RequestAccess doesn't continue.

This is the code in question, which works fine on Android:

```
       var push = Host.Current.Services.GetService<IPushManager>();
       var result = await push.RequestAccess();
    // never reaches this part
         Console.WriteLine($"Push access result: {result.Status}");

```

remote-notifications are ok and the GoogleService-Info.plist is also added.
I would expect an error when these are not ok, is that correct?
Does anyone have another idea how to fix this?

Console doesn't show any valuable information as far as I can see.

[0:] ShinyStartup: Information: Startup Binding for Shiny.Push.PushManager
[0:] Shiny.Stores.Impl.ObjectStoreBinder: Information: Success: Shiny.Push.PushManager to store: settings
[..]
[0:] Shiny.Hosting.IosLifecycleExecutor: Information: Shiny.Push.PushManager handling lifecycle event for Shiny.Hosting.IIosLifecycle+IOnFinishedLaunching

I'm testing in Debug mode with a physical IOS device.

This is also null:
 Host.Current.Services.GetService<IPushManager>().RegistrationToken

**Comment by stefandenhartog** [comment]
Just did a better search and came accross this https://shinylib.net/client/push/providers/

Trying it out now.
Issue is the same with   
`builder.Services.AddPushFirebaseMessaging<PushDelegate>();`

Any idea would be great :)

**Reply by aritchie** [reply]
Firebase on iOS needs a vastly updated binding which I haven't ported over yet.  Native iOS & azure notification hubs work just fine though.  If you're having problems with either of those, please file a reproducible sample in the issues section
